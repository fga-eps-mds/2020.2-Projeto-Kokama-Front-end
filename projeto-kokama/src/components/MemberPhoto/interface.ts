export interface MemberPhotoProps {
  name: string;
  photo: any;
}
