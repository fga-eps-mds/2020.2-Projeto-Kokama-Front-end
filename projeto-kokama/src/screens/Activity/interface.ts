export interface Exercise {
    id: number;
    phrase_portuguese: string;
    phrase_kokama: string;
    options: Array<string>;
}