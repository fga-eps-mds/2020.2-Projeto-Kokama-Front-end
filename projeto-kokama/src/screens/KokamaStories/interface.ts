export interface kokamaStories {
    title: string;
    text: string;
}