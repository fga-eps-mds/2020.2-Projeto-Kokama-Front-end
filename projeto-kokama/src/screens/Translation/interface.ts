export interface Phrase {
    phrase_portuguese: string;
    phrase_kokama: string;
}

export default interface Dictionary {
    word_kokama: string;
    pronunciation_type: string;
    phrases: Phrase[];
    translations: string[];
}