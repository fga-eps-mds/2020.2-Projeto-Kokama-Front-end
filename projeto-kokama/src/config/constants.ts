export const PORTUGUESE = "PortuguÃªs";
export const KOKAMA = "Kokama";

export const HISTORYSIZE = 10;