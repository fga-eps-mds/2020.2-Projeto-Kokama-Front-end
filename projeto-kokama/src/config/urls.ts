import {TRANSLATE_MICROSERVICE_URL} from "@env";

export const TRANSLATE_URL = TRANSLATE_MICROSERVICE_URL;