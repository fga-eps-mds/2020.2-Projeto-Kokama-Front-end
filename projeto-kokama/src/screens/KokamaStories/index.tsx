import {
    View,
    Text,
    ScrollView,
    SafeAreaView,
    TouchableWithoutFeedback
} from "react-native";
import React, { useEffect, useState } from "react";
import styles from "./styles";
import { KokamaStories } from "./interface";
import Api from "../../api/Api";
import SpinnerLoading from "../../components/SpinnerLoading";
import Icon from "react-native-vector-icons/AntDesign";
import {
    PORTUGUESE,
    KOKAMA,
} from "../../config/constants";
import StoryList from "../../components/StoryList";
import { SearchBar } from 'react-native-elements';
import { LEARN_MICROSERVICE_URL } from "@env"


export default function Stories() {
    const [kokamaStories, setKokamaStories] = useState<Array<KokamaStories>>();
    const [originLanguage, setOriginLanguage] = useState(PORTUGUESE);
    const [search, setSearch] = useState("");
    if (originLanguage == "") {
        setOriginLanguage(PORTUGUESE);
    }

    useEffect(() => {
        const fetchData = async () => {
            const result = await Api(
                LEARN_MICROSERVICE_URL + "ensino/lista_de_historias/?format=json"
            );
            if (result.status === 200 && kokamaStories != result.data) {
                setKokamaStories(result.data);
            } else {
                console.log("A requisição não pôde ser concluída.\n[Status: ", result.status, "]");
            }
        }
        fetchData();
    }, []);

    function exchangeLanguage() {
        if (originLanguage === PORTUGUESE) {
            setOriginLanguage(KOKAMA);
        } else {
            setOriginLanguage(PORTUGUESE);
        }
    }

    return (
        <SafeAreaView>
            {kokamaStories && kokamaStories.length == 0 && (
                <SpinnerLoading />
            )}
            {kokamaStories && kokamaStories.length > 0 && (
                <ScrollView style={styles.container} keyboardShouldPersistTaps="always">
                    <View style={styles.searchBarArea}>
                        <SearchBar
                            placeholder="Pesquise aqui..."
                            onChangeText={setSearch}
                            value={search}
                            platform="android"
                            containerStyle={ styles.searchBar }
                        />

                        <View style={styles.swapButtonArea}>
                            <TouchableWithoutFeedback onPress={exchangeLanguage}>
                                <Icon name="swap" size={40} />
                            </TouchableWithoutFeedback>
                            <Text>{originLanguage}</Text>
                        </View>
                    </View>
                    <StoryList
                        search={search}
                        list={kokamaStories}
                        language={originLanguage || ""}
                    />
                </ScrollView>
            )}
        </SafeAreaView>
    );
}