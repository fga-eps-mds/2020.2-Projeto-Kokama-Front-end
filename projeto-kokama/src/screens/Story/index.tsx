import {
    View,
    Text,
    ScrollView,
    SafeAreaView,
} from "react-native";
import React from "react";
import styles from "./styles";

export default function Story({ route, navigation }) {
    const story = route.params['story'];
    interface Props {
        language: string;
    }

    return( 
        <SafeAreaView>
            <ScrollView style={styles.container}>
                <View style={styles.container}> 
                    {route.params.language == "Kokama" && (
                        <View>
                            <Text style={styles.titleField}>
                                {story.title_kokama}
                            </Text>
                            <Text style={styles.textField}>
                                {story.text_kokama}
                            </Text>
                            <Text style={styles.titleField}>
                                {story.title_portuguese}
                            </Text>
                            <Text style={styles.textField}>
                                {story.text_portuguese}
                            </Text>
                        </View>
                    ) || (
                        <View>
                            <Text style={styles.titleField}>
                                {story.title_portuguese}
                            </Text>
                            <Text style={styles.textField}>
                                {story.text_portuguese}
                            </Text>
                            <Text style={styles.titleField}>
                                {story.title_kokama}
                            </Text>
                            <Text style={styles.textField}>
                                {story.text_kokama}
                            </Text>
                        </View>
                    )}
                </View>
            </ScrollView>
        </SafeAreaView>
    );
}