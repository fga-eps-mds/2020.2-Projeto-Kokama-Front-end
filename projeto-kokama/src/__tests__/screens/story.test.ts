import renderer from 'react-test-renderer';
import Story from '../../screens/Story';
import { KokamaStories } from '../../screens/KokamaStories/interface';
import { KOKAMA, PORTUGUESE } from '../../config/constants';


test("Check Story screen", () => {
    const expected: KokamaStories =
    {
        id: 2,
        title_portuguese: 'outro titulo portugues',
        text_portuguese: 'outro texto portugues',
        title_kokama: 'outro titulo kokama',
        text_kokama: 'outro texto kokama'
    }
    let params = {'story': expected, 'language': KOKAMA}
    let story = Story({ route: {'params': params}, navigation: null})
    let tree = renderer.create(story)
    expect(tree).toMatchSnapshot();

    params = {'story': expected, 'language': PORTUGUESE}
    story = Story({ route: {'params': params}, navigation: null})
    tree = renderer.create(story)
    expect(tree).toMatchSnapshot();
})