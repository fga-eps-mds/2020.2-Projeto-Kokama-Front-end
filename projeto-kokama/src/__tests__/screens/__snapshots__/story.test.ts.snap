// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Story screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "height": 1334,
        "maxWidth": 750,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FAFAFA",
            "height": 1334,
            "maxWidth": 750,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginTop": 10,
                "padding": 10,
                "textAlign": "center",
              }
            }
          >
            outro titulo kokama
          </Text>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginBottom": 20,
                "padding": 15,
                "textAlignVertical": "center",
              }
            }
          >
            outro texto kokama
          </Text>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginTop": 10,
                "padding": 10,
                "textAlign": "center",
              }
            }
          >
            outro titulo portugues
          </Text>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginBottom": 20,
                "padding": 15,
                "textAlignVertical": "center",
              }
            }
          >
            outro texto portugues
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Check Story screen 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "height": 1334,
        "maxWidth": 750,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FAFAFA",
            "height": 1334,
            "maxWidth": 750,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginTop": 10,
                "padding": 10,
                "textAlign": "center",
              }
            }
          >
            outro titulo portugues
          </Text>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginBottom": 20,
                "padding": 15,
                "textAlignVertical": "center",
              }
            }
          >
            outro texto portugues
          </Text>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginTop": 10,
                "padding": 10,
                "textAlign": "center",
              }
            }
          >
            outro titulo kokama
          </Text>
          <Text
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "fontSize": 20,
                "marginBottom": 20,
                "padding": 15,
                "textAlignVertical": "center",
              }
            }
          >
            outro texto kokama
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
